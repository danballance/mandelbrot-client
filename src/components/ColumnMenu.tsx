import * as React from "react";
import { Container, Grid, Button } from 'semantic-ui-react'

interface Props {
    onZoomInClicked: (event: React.MouseEvent<HTMLButtonElement>) => void;
    onZoomOutClicked: (event: React.MouseEvent<HTMLButtonElement>) => void;
    onResetClicked: (event: React.MouseEvent<HTMLElement>) => void;
}

export const ColumnMenu: React.FC<Props> = (props: Props): JSX.Element => (
    <Grid.Column mobile={16} computer={8} textAlign="left">
        <Container className="controls">
            <Button.Group>
                <Button 
                    id="zoom-in" 
                    inverted 
                    circular 
                    icon="zoom-in" 
                    onClick={props.onZoomInClicked} />
                <Button 
                    id="zoom-out" 
                    inverted 
                    circular 
                    icon="zoom-out" 
                    onClick={props.onZoomOutClicked} />
            </Button.Group>
            <Button
                id="zoom-reset" 
                inverted 
                icon="fast backward"
                content="Reset" 
                onClick={props.onResetClicked} />    
        </Container>
    </Grid.Column>
)
