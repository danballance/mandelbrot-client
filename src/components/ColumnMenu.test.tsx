import React from 'react';
import { shallow } from 'enzyme'
import { ColumnMenu } from './ColumnMenu'

const onResetClicked = jest.fn();
const onZoomInClicked = jest.fn();
const onZoomOutClicked = jest.fn();

const columnMenu = shallow(<ColumnMenu
    onResetClicked={onResetClicked}
    onZoomInClicked={onZoomInClicked}
    onZoomOutClicked={onZoomOutClicked} 
/>);

describe('ColumnMenu', (): void => {
    it('trigggers onZoomInClicked when zoom in button clicked', (): void => {
        columnMenu
            .find('Container.controls ButtonGroup Button#zoom-in')
            .simulate('click');
        expect(onZoomInClicked).toHaveBeenCalled();
    })

    it('trigggers onZoomOutClicked when zoom out button clicked', (): void => {
        columnMenu
            .find('Container.controls ButtonGroup Button#zoom-out')
            .simulate('click');
        expect(onZoomOutClicked).toHaveBeenCalled();
    })

    it('trigggers onResetClicked when zoom reset button clicked', (): void => {
        columnMenu
            .find('Container.controls Button#zoom-reset')
            .simulate('click');
        expect(onResetClicked).toHaveBeenCalled();
    })

    it('renders correctly', (): void => {
        expect(columnMenu).toMatchSnapshot();
    })
});
