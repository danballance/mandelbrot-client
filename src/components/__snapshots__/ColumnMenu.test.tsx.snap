// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColumnMenu renders correctly 1`] = `ShallowWrapper {}`;
